src/popup/router/components/Modal.vue

```diff
@@ -1,33 +1,49 @@
 <template>
-  <transition appear>
+  <transition
+    appear
+    :name="fromBottom ? 'from-bottom-transition' : 'pop-in-transition'"
+  >
     <div
       class="modal"
-      :class="{'full-screen': fullScreen}"
+      :class="{
+        'full-screen': fullScreen,
+        'from-bottom': fromBottom,
+        'has-close-button': hasCloseButton,
+      }"
     >
-      <div class="container">
-        <ButtonPlain
-          v-if="close"
-          class="close"
+      <div
+        v-click-outside="() => $emit('close')"
```

Comment:
Can we do it like that?
```suggestion
        v-click-outside="$emit('close')"
```

Reply:
No, because this way the modal closes immediately after opening it. Custom directives works little bit different than the native ones. Native ones tries to match the component's method and eval it.